frontend/src/components/IntegrationConfiguration.tsx

```diff
@@ -218,49 +225,59 @@ const SelectBoard: FC<{ configuration: IntegrationConfiguration }> = ({
+          {(tokenError || boardsTokenError || errorMessage) && (
             <Alert
               severity={errorMessage ? 'error' : 'info'}
               onClose={() => {
-                if (tokenError) setTokenError(false);
+                if (boardsTokenError) setBoardsTokenError(false);
+                if (tokenError) {
+                  boards.refetch();
+                  selected.refetch();
+                  setTokenError(false);
+                }
                 if (errorMessage) setErrorMessage('');
               }}
               icon={false}
             >
               {errorMessage || t('Oauth can not select board')}
             </Alert>
           )}
-          <Select
-            name="board"
-            id="board"
-            className="react-select"
-            classNamePrefix="react-select"
-            styles={{ menuPortal: (base) => ({ ...base, zIndex: 9999 }) }}
-            placeholder="Select board"
-            isDisabled={boards.isError}
-            menuPortalTarget={document.body}
-            onChange={(entry) => {
-              if (entry) {
-                patch({ id, name, roadmapId, boardId: entry.value })
-                  .unwrap()
-                  .catch((err) => {
-                    setErrorMessage(
-                      err.data?.message ?? 'something went wrong',
-                    );
-                  });
-              }
-            }}
-            defaultValue={
-              selected.data && {
-                value: selected.data.id,
-                label: selected.data.name,
+          {!selected.isError && (
+            <Select
+              name="board"
+              id="board"
+              className="react-select"
+              classNamePrefix="react-select"
+              styles={{ menuPortal: (base) => ({ ...base, zIndex: 9999 }) }}
+              placeholder="Select board"
+              isDisabled={boards.isError}
+              menuPortalTarget={document.body}
+              onChange={(entry) => {
+                if (entry) {
+                  patch({ id, name, roadmapId, boardId: entry.value })
+                    .unwrap()
+                    .catch((err) => {
+                      setErrorMessage(
+                        err.data?.message ?? 'something went wrong',
+                      );
+                    });
+                }
+              }}
+              defaultValue={
+                selected.data && {
+                  value: selected.data.id,
+                  label: selected.data.name,
+                }
               }
-            }
-            options={boards.data?.map((board) => ({
-              value: board.id,
-              label: board.name,
-            }))}
-          />
+              options={boards.data?.map((board) => ({
+                value: board.id,
+                label: board.name,
+              }))}
+            />
+          )}
+          {boardId && !boards.isError && !selected.isError && (
```

Comment:
```suggestion
          {boardId && selected.isSuccess && (
```

This shouldn't require all boards, only that a board is selected. The selected board query could work as a "token refresher" before rendering this, if the refresh would be added to that on the server